### PR TITLE
+ correctly expose `methods` option to recover compatibility to `vue-class-component`

### PR DESCRIPTION
**Pull request by tall1on**
Hey, I noticed that the `methods` option doesn't work like the one from `vue-class-component`, so I added it. I also improved a bit of other code. Tests have all been passed. I also edited the `methods` test to make sure it keeps working like in `vue-class-component`.

### CODE REVIEW

**Comment by fb-sean** [comment]
Really needed thanks for the PR!

File: src/component.ts [hunk 1]
```diff
@@ -92,11 +93,17 @@ function buildComponent(cons: Cons, arg: ComponentOption, extend?: any): any {
     }
     option.emits = emits
 
+    //apply methods
```
Comment (ruojianll) [inline src/component.ts:96]: Methods in class has already applied into `optoin.methods`, we just merge `arg.methods`.
```
if ('object' === typeof arg.methods && !Array.isArray(arg.methods) && arg.methods !== null) {
        option.methods??={}
        Object.assign(option.methods, arg.methods);
    }
```

File: src/option/emit.ts
Comment (ruojianll) [inline src/option/emit.ts:1]: `methods` in slot is not necessary, so we don't change this file.

File: src/option/methodsAndHooks.ts
Comment (ruojianll) [inline src/option/methodsAndHooks.ts:1]: Remove all codes about slot's `mehtods`

File: src/utils.ts [hunk 2]
```diff
@@ -28,6 +28,7 @@ export type SlotMapTypes = {
     ref: Map<string, RefConfig>
     setup: Map<string, SetupConfig>
     customDecorator: Map<string, CustomDecoratorRecord[]>
+    methods: Map<string, Function>
```
Comment (ruojianll) [inline src/utils.ts:31]: We don't need this since we changed codes in `component.ts`.

**Review by ruojianll — CHANGES_REQUESTED** [review]
I left some messages and hope you can improve it.

**Comment by tall1on** [comment]
I guess i overthought it a bit 😅

I changed everything you asked, all tests passed.

**Review by ruojianll — APPROVED** [review]

**Pull request merged**